### PR TITLE
setup basic logging for native

### PR DESCRIPTION
**Pull request by aajtodd**
*Issue #, if available:*

*Description of changes:*
Add log subject for aws-crt-kotlin (native)
Add logging of errors to several callbacks

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt [hunk 1]
```diff
@@ -68,3 +68,7 @@ internal object Logging {
         }
     }
 }
+
+internal inline fun klog(level: LogLevel, message: String, subject: aws_log_subject_t = AWS_LS_KOTLIN_CRT_GENERAL) {
+    s_crt_kotlin_log(level.value.convert(), subject, message)
+}
```
Comment (ianbotsf) [inline aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt:74]: **Style**: `klog` is not the sort of identifier we'd typically use in Kotlin code. Is the `k` to differentiate from some other `log` method? Is there a reason for this not to be part of the `Logging` object?

**Review by ianbotsf — APPROVED** [review]

File: aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt (hunk 1 above)
Comment (aajtodd) [reply aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt:74]: It was just a short hand for identifying that we are targeting the `AWS_LS_KOTLIN_CRT_GENERAL` subject (i.e. logging a kotlin related logging message to CRT logger as opposed to some other subject). 

I don't care what we name it, we could just shorten it to `log` if you prefer.

Comment (lauzadis) [reply aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt:74]: I like `log`

**Review by lauzadis — APPROVED** [review]

File: aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt (hunk 1 above)
Comment (ianbotsf) [reply aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt]: I _think_ `log` makes sense for the majority of our logging needs. _Will_ we be logging from **aws-crt-kotlin** to a subject other than `AWS_LS_KOTLIN_CRT_GENERAL`?

**Pull request merged**